SparkPost/matchbox
# UX-505 Upgrade build dependencies

### What Changed
- Updates some build time dependencies for matchbox and matchboxi-icons
- Removes legacy and unused dependencies for matchbox

### How To Test or Verify
- Run the matchbox and matchbox-icons builds separately
- Ensure both builds pass

Test the build locally:
- Remove `/src` from the libby webpack alias list for matchbox and matchbox icons
```js
// libby.config.js
alias: {
  '@sparkpost/matchbox-icons': '/packages/matchbox-icons',
  '@sparkpost/matchbox': '/packages/matchbox',
  '@sparkpost/design-tokens': '/packages/design-tokens',
  '@sparkpost/matchbox-media': '/packages/matchbox-media',
},
```
- Run libby
- Verify the stories work as expected

### PR Checklist
